api/src/test/java/org/openmrs/api/db/hibernate/HibernateOrderDAOTest.java

```diff
@@ -65,4 +74,82 @@ public void saveOrderGroup_shouldSaveOrderGroup() {
 		}
 		
 	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByEncounter_shouldFailGivenNullEncounter() {
+		dao.getOrderGroupsByEncounter(null);
+	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByPatient(Patient)}
+	 * @throws Exception
+	 */
+	@Test(expected = APIException.class)
+	public void getOrderGroupsByPatient_shouldFailGivenNullPatient() {
+		dao.getOrderGroupsByPatient(null);
+	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFromAnEncounter() {
+		Encounter existingEncounter1 = Context.getEncounterService().getEncounter(4);
+		List<OrderGroup> ordergroups1 = Context.getOrderService().getOrderGroupsByEncounter(existingEncounter1);
+		
+		assertEquals(1, ordergroups1.size());
+	}
+	
+	/**
+	 * @see {@link HibernateOrderDAO#getOrderGroupsByEncounter(Encounter)}
+	 * @throws Exception
+	 */
+	@Test
+	public void getOrderGroupsByEncounter_shouldGetOrderGroupsFrommMultipleEncounters() {
+		Encounter existingEncounter1 = Context.getEncounterService().getEncounter(3);
+		Encounter existingEncounter2 = Context.getEncounterService().getEncounter(4);
```

Comment:
Is there any reason why we are repeating what we have already done in the test before? That is the fetching of order groups by encounterId = 4

Reply:
@dkayiwa according the dataset below in this pr i thought you can various tests in different test cases  on the same dataset,
i stand to be corrected on this